modules/stack/main.tf

```diff
@@ -21,6 +21,24 @@ resource "spacelift_stack" "default" {
   terraform_workspace = var.terraform_workspace
 }
 
+resource "spacelift_run" "this" {
+  count = var.enabled && var.commit_sha != null ? 1 : 0
+
+  stack_id   = spacelift_stack.default[0].id
+  commit_sha = var.commit_sha
+
+  keepers = {
+    branch = spacelift_stack.default[0].branch
+  }
```

Comment:
The branch will likely not change so we could remove these keepers. We do not need a keeper on the `commit_sha` itself.